spec/views/hq/users/index_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+describe "hq/users/index.html.erb", type: :view do
+  it 'should display a valid user' do
+    assign(:users, [ FactoryGirl.build_stubbed(:user, user_name: 'User1', email: 'Email1@example.com') ] )
+    render 
+    expect(rendered).to match /User1/
+    expect(rendered).to match /Email1@example.com/
+  end
+
+  it 'should have a New User link' do
+    assign(:users, [])
+    render
+    expect(rendered).to have_link('New User', new_hq_user_path)
+  end
+
+  describe 'should display assigned active sponsors correctly' do
+
+    context 'no active sponsors' do
+      it 'displays 0 active sponsors' do
+        @users = [ instance_double(User, active_sponsors: [], user_name: 'usr', email: 'email@e.e') ]
+        render
+        expect(rendered).to have_selector('td', text: '0')
```

Comment:
There are some things in here I'm not comfortable with, Nik addressed a lot of them.  I would add to his comment though that `@users.each do` is a code fork so we should probably ensure that it correctly handles 0, 1, and more than one enumerable items in @users that all respond to `.id`, `.user_name`, `.email`, and `.active_sponsors.count`.  An array of instance_doubles of User fits perfectly.  I wouldn't check for certain tags or anything that's outside the scope of what this view does.  The data coming from `active_sponsors.count` is an example of that.  If you feed it 2 users, does it give you back 2 links to those users' show pages? How about 1? Zero @users?  That would test all of the code forks (in osra331's index view)